### PR TITLE
Added support for extending handshake metadata

### PR DESCRIPTION
**Pull request by rozza**
JAVARX-29

### CODE REVIEW

File: driver/src/main/com/mongodb/rx/client/MongoClients.java [hunk 1]
```diff
@@ -99,7 +107,74 @@ public static MongoClient create(final ConnectionString connectionString) {
...
+     * @return the client
+     * @since 1.3
+     */
+    public static MongoClient create(final MongoClientSettings settings, final ObservableAdapter observableAdapter,
+                                     final MongoDriverInformation mongoDriverInformation) {
+        return new MongoClientImpl(com.mongodb.async.client.MongoClients.create(settings,
+                updateMongoDriverInformation(mongoDriverInformation)), observableAdapter);
+    }
+
+    private static MongoDriverInformation updateMongoDriverInformation(final MongoDriverInformation mongoDriverInformation) {
```
Comment (jyemin) [inline driver/src/main/com/mongodb/rx/client/MongoClients.java]: Can this happen just once rather than on ever MongoClient creation?

Comment (rozza) [inline driver/src/main/com/mongodb/rx/client/MongoClients.java]: No - it depends if someone passes in any MongoDriverInformation we have to extend it. Same as the java driver.

Comment (jyemin) [inline driver/src/main/com/mongodb/rx/client/MongoClients.java]: OK, I see.  My point was more that getting the driver version from the manifest could be cached.

On the flip side, there's no reason to have a static MongoDriverInformation instance.  There's no real penalty for creating that anew each time.

Comment (rozza) [inline driver/src/main/com/mongodb/rx/client/MongoClients.java]: Yeah its just to keep the code dry - as the default is used in multiple places and the updating is done in multiple places. Mind you `DEFAULT_DRIVER_INFORMATION` is slightly misleading.  But as the builder prepends the information it needs to apply the settings to the user defined settings.

Comment (rozza) [inline driver/src/main/com/mongodb/rx/client/MongoClients.java]: Updated - to clarify the intent and cache the settings.

**Pull request merged**